src/scene_title.cpp

```diff
@@ -51,9 +51,9 @@ void Scene_Title::Start() {
 }
 
 void Scene_Title::Continue() {
-	if (Game_Temp::restart_title_cache) {
-		// Clear the cache when the game returns to the
-		// title screen e.g. by pressing F12, except the Title Load menu
+	if (restart_title_cache) {
+		// Clear the cache when the game returns to the title screen
+		// e.g. by pressing F12, except the Title Load menu
 		Cache::Clear();
 		AudioSeCache::Clear();
```

Comment:
Does this logic have to be done in `Scene_Title::Continue()`? Possible to just do it directly in `OnGameLoad()` (rename it) and then call that directly on startup also?

We localized the restart flag which is good. Possible to eliminate it entirely?

Reply:
Actually I see how this works now. Doing it this way probably makes sense to ensure we don't miss an edge case.

Reply:
It should be possible to put some of this tear-down in the Scene_Map deconstructor, this way we don't depend on the Title scene for this but I don't see a simply way to get rid of this boolean for the "Don't restart BGM when returning from Load scene" edge case.

Reply:
Better to add state to handle it than try to juggle the edge cases. I was originally mistaken. The way you did it here is probably best.